core/state/pruner/pruner.go

```diff
@@ -188,8 +188,15 @@ func prune(maindb ethdb.Database, stateBloom *stateBloom, middleStateRoots map[c
 		cstart := time.Now()
 
 		for b := byte(0); b < byte(16); b++ {
-			log.Info("Compacting database", "range", fmt.Sprintf("%#x-%#x", b, b+1), "elapsed", common.PrettyDuration(time.Since(cstart)))
-			if err := maindb.Compact([]byte{b}, []byte{b + 1}); err != nil {
+			var (
+				start = []byte{b << 4}
+				end   = []byte{(b+1)<<4 - 1}
```

Comment:
I don't think this is correct either. You'll have e.g. start `0x40` and end `0x4f`. But the `end` is the `limit` in the leveldb range, which is _exclusive_. 

I think.

Reply:
Bleah, you're right.

```
	// Limit of the key range, not include in the range.
	Limit []byte
```

Myeah, wanna fix it? :)

Reply:
Sure